src/Ryujinx.Memory/MemoryManagementUnix.cs

```diff
@@ -186,17 +186,21 @@ public unsafe static IntPtr CreateSharedMemory(ulong size, bool reserve)
                 throw new OutOfMemoryException();
             }
 
-            return (IntPtr)fd;
+            return fd;
         }
 
         public static void DestroySharedMemory(IntPtr handle)
         {
+#pragma warning disable CA2020 // Prevent behavioral change caused by built-in operators of IntPtr/UIntPtr
             close((int)handle);
```

Comment:
```suggestion
            close(handle.ToInt32());
```
And remove the pragma